app/src/main/java/com/android/unio/ui/map/Map.kt

```diff
@@ -1,14 +1,162 @@
+import com.google.firebase.Timestamp
+import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.Marker
+import com.google.maps.android.compose.MarkerState
+import com.google.maps.android.compose.rememberCameraPositionState
+import java.util.Calendar
+import java.util.concurrent.TimeUnit
 
-@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun MapScreen() {
-  Scaffold(modifier = Modifier.testTag("MapScreen")) { Text("Map screen") }
+fun MapScreen(
+    navigationAction: NavigationAction,
+    eventListViewModel: EventListViewModel = viewModel(factory = EventListViewModel.Factory)
+) {
+  val context = LocalContext.current
+  Scaffold(
+      modifier = Modifier.testTag("MapScreen"),
+      topBar = {
+        TopAppBar(
+            title = {
+              Text(
+                  context.getString(R.string.map_event_title),
+                  modifier = Modifier.testTag("MapTitle"))
+            },
+            navigationIcon = {
+              IconButton(
+                  onClick = { navigationAction.goBack() },
+                  modifier = Modifier.testTag("goBackButton")) {
+                    Icon(
+                        imageVector = Icons.AutoMirrored.Outlined.ArrowBack,
+                        contentDescription = context.getString(R.string.map_event_go_back_button))
+                  }
+            })
+      }) { pd ->
+        EventMap(pd, eventListViewModel)
+      }
+}
+
+@Composable
+fun EventMap(pd: PaddingValues, eventListViewModel: EventListViewModel) {
+  val cameraPositionState = rememberCameraPositionState {
+    position = CameraPosition.fromLatLngZoom(LatLng(46.518831258, 6.559331096), 10f)
+  }
+
+  val events = eventListViewModel.events.collectAsState()
+
+  /** Mock arbitrary saved event for all users. */
+  val arbitrarySavedEvents: List<Event> =
+      listOf(
+          Event(
+              uid = "123456789",
+              title = "Arbitrary Saved Event",
+              organisers = Association.emptyFirestoreReferenceList(),
+              taggedAssociations = Association.emptyFirestoreReferenceList(),
+              description = "This is the description of an arbitrary saved event.",
+              catchyDescription = "This is an arbitrary saved event.",
+              date =
+                  Timestamp(
+                      Calendar.getInstance()
+                          .apply { add(Calendar.DAY_OF_YEAR, 1) }
+                          .time), // tomorrow's date
+              location =
+                  Location(
+                      latitude = 46.51848436506024, longitude = 6.568259761045008), // Rolex Center
+          ))
+
+  GoogleMap(
+      modifier = Modifier.padding(pd).testTag("googleMaps"),
+      cameraPositionState = cameraPositionState) {
+        // Display saved events
+        arbitrarySavedEvents.forEach { event ->
+          if (event.date.toDate() > Calendar.getInstance().time) {
```

Comment:
You're right, but that would mean deleting old events. I think it will be necessary to keep some of the old events in the database and remove them later, as we will potentially add some social features for past events.

Reply:
Yes absolutely. As I said i was nit picking and this is completly fine as it's done right now.